packages/e2e-tests/specs/editor/blocks/post-title.test.js

```diff
@@ -13,23 +13,18 @@ describe( 'Post Title block', () => {
 	} );
 
 	it( 'Can edit the post title', async () => {
-		// Create a block with some text that will trigger a list creation.
 		await insertBlock( 'Post Title' );
-		const editablePostTitleSelector =
-			'.wp-block-post-title[contenteditable="true"]';
-		await page.waitForSelector( editablePostTitleSelector );
-		await page.focus( editablePostTitleSelector );
 
-		// Create a second list item.
-		await page.keyboard.type( 'Just tweaking the post title' );
+		const titleText = 'Just tweaking the post title';
+
+		await page.click( '[aria-label="Block: Post Title"]' );
+		await page.keyboard.type( titleText );
 
 		await saveDraft();
 		await page.reload();
-		await page.waitForSelector( '.edit-post-layout' );
-		const title = await page.$eval(
-			'.editor-post-title__input',
-			( element ) => element.textContent
+
+		await page.waitForSelector(
+			`.editor-post-title >> text="${ titleText }"`
 		);
-		expect( title ).toEqual( 'Just tweaking the post title' );
```

Comment:
Because IMO it's the equivalent of [the selector above](https://github.com/WordPress/gutenberg/blob/8080f31191b95aac0cd3ededcb85b299e7779ea8/packages/e2e-tests/specs/editor/blocks/post-title.test.js#L26-L28):
```js
await page.waitForSelector(
	`.editor-post-title >> text="${ titleText }"`
);
```

Reply:
Maybe it was just there for readability. `waitForSelector` isn't really an assertion.

Off-topic, but IMO, a better solution would be to have an assertion **explicitly** for querying elements. Something like:

```js
await expect( `.editor-post-title >> text="${ titleText }"` ).toBeFoundIn( page );
```

`puppeteer-testing-library` has something similar OOTB, but we can always create something similar for whatever querying libraries we use.

Reply:
> Maybe it was just there for readability. waitForSelector isn't really an assertion.

True, although if we use the inner text as the selector, the assertion will never be reached in case that text is not present.